src/core/unittest/TicketTest.cpp

```diff
@@ -578,8 +582,443 @@ TEST(ResumptionTicketTest, ServerEncDecNoAppData)
+        QUIC_TP_FLAG_INITIAL_MAX_STRM_DATA_UNI |
+        QUIC_TP_FLAG_INITIAL_MAX_STRMS_BIDI |
+        QUIC_TP_FLAG_INITIAL_MAX_STRMS_UNI;
+    ServerTP.ActiveConnectionIdLimit = QUIC_TP_ACTIVE_CONNECTION_ID_LIMIT_MIN;
+
+    // Set IPv4 address 192.0.2.1 (do not set port)
+    CxPlatZeroMemory(&CarefulResumeState.RemoteEndpoint, sizeof(CarefulResumeState.RemoteEndpoint));
+    CarefulResumeState.RemoteEndpoint.Ipv4.sin_family = QUIC_ADDRESS_FAMILY_INET;
+    CarefulResumeState.RemoteEndpoint.Ipv4.sin_addr.s_addr = htonl(0xC0000201); // 192.0.2.1
+
+    // Test all valid QUIC_CONGESTION_CONTROL_ALGORITHM values
+    const struct {
+        QUIC_CONGESTION_CONTROL_ALGORITHM Algorithm;
+        const char* Name;
+    } kAlgorithms[] = {
+        { QUIC_CONGESTION_CONTROL_ALGORITHM_CUBIC, "CUBIC" },
+#if defined(QUIC_API_ENABLE_PREVIEW_FEATURES)
+        { QUIC_CONGESTION_CONTROL_ALGORITHM_BBR, "BBR" },
+#endif
+    };
+
+    for (size_t i = 0; i < ARRAYSIZE(kAlgorithms); ++i) {
+        // Populate CarefulResumeState with test values
+        CarefulResumeState.SmoothedRtt = 12345 + (uint64_t)i;
+        CarefulResumeState.MinRtt = 2345 + (uint64_t)i;
+        CarefulResumeState.Expiration = 0x1122334455667788 + (uint64_t)i;
+        CarefulResumeState.Algorithm = kAlgorithms[i].Algorithm;
+        CarefulResumeState.CongestionWindow = 65536 + (uint32_t)i;
+
+        EncodedServerTicket = nullptr;
+        EncodedServerTicketLength = 0;
+        CxPlatZeroMemory(&DecodedServerTP, sizeof(DecodedServerTP));
+        CxPlatZeroMemory(&DecodedCarefulResumeState, sizeof(DecodedCarefulResumeState));
+        DecodedAppData = nullptr;
+        DecodedAppDataLength = 0;
+
+        TEST_QUIC_SUCCEEDED(
+            QuicCryptoEncodeServerTicket(
+                nullptr,
+                QUIC_VERSION_LATEST,
+                0,
+                nullptr,
+                &ServerTP,
+                &CarefulResumeState,
+                NegotiatedAlpn[0],
+                NegotiatedAlpn + 1,
+                &EncodedServerTicket,
+                &EncodedServerTicketLength));
+
+        ASSERT_NE(EncodedServerTicket, nullptr);
+        ASSERT_NE((uint16_t)EncodedServerTicketLength, 0);
+
+        TEST_QUIC_SUCCEEDED(
+            QuicCryptoDecodeServerTicket(
+                &Connection,
+                (uint16_t)EncodedServerTicketLength,
+                EncodedServerTicket,
+                NegotiatedAlpn,
+                sizeof(NegotiatedAlpn),
+                &DecodedServerTP,
+                &DecodedCarefulResumeState,
+                &DecodedAppData,
+                &DecodedAppDataLength));
+
+        ASSERT_EQ((uint16_t)DecodedAppDataLength, 0);
+        ASSERT_EQ(DecodedAppData, nullptr);
+        CompareTransportParameters(&ServerTP, &DecodedServerTP);
+
+        // Validate CarefulResumeState fields (except port)
+        ASSERT_EQ(CarefulResumeState.SmoothedRtt, DecodedCarefulResumeState.SmoothedRtt) << kAlgorithms[i].Name;
```

Comment:
I have not seen this syntax, ` << kAlgorithms[i].Name;`. What does it do?

Reply:
Ok, JFYI. I am not sure if we compile unittest in kernel mode. If we do, this won't be compiled.